### ISSUE TITLE
allow copying to clipboard instead of saving as a file

### ISSUE THREAD

**Issue by luisrudge** [octref/polacode#9]

**Comment by ericcirone** [comment]
Just tried out Polacode and while it is cool, this was my immediate thought as well.

**Comment by nodeful** [comment]
Absolutely, a must have functionality!

**Comment by octref** [comment]
Yep, had it in mind but didn't make it for MVP.

**Comment by octref** [comment]
This seems impossible with the current API. The Electron webview can't do it per this [Chromium Bug](https://bugs.chromium.org/p/chromium/issues/detail?id=150835). On macOS, pbcopy does not take image input. Electron does have an API but VS Code does not expose it.

I'm asking for it: https://github.com/Microsoft/vscode/issues/217#issuecomment-366495874

**Comment by seveves** [comment]
@octref I tried to implement it this with the electron nativeImage + clipboard modules but can't get it running. I think we have to wait for Microsoft to expose the clipboard API via vscode.

**Comment by sangeeth96** [comment]
I don't think this solves this issue but can we have like a toggle to preview the image and right-click that and have Chrome's copy image functionality? Is that possible in VS Code right now? I checked the image previewer in Code but that doesn't allow me to copy the image being previewed.

**Comment by kamranayub** [comment]
Random thought: has anyone tried using Canvas?

From image, use [drawImage](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/drawImage). Then save using `toDataUrl`:

https://stackoverflow.com/questions/10673122/how-to-save-canvas-as-an-image-with-canvas-todataurl

**Comment by sangeeth96** [comment]
@kamranayub On the same lines as I was wondering which I posted above but right-clicking doesn't seem to popup any context menus inside VS Code.

**Comment by travisbaratcart** [comment]
+1, I want to use this extension to copy paste code into emails. I'm not going to go through saving a file and importing for that.

**Comment by adi518** [comment]
Someone made a fork that fixes a bunch of things, so I'd use that for the time being.

**Comment by duartealexf** [comment]
duplicate of #70? I'd really like this feature too!

**Comment by kufii** [comment]
I created an npm library for copying images to clipboard that you're free to use:

https://www.npmjs.com/package/img-clipboard

**Comment by JounQin** [comment]
This is definitely cool feature, will raise an PR for it when I'm not so busy.

**Comment by GalDayan** [comment]
Awesome feature, very useful

**Comment by MathBunny** [comment]
Hi, any updates on this?

**Comment by mreduar** [comment]
I found an extension that already implements this very useful feature.
https://marketplace.visualstudio.com/items?itemName=adpyke.codesnap
